### ISSUE TITLE
missing collection

### ISSUE THREAD

**Issue by GabrielJones646** [SoftInstigate/restheart#78]
The db page lists several collections.
When I visit the various collection pages it says 404 Not Found.

v1.1.3-SNAPSHOT
mongo v3.0.7

I'll do some debugging later today to see if I can pinpoint the bug.

**Comment by ujibang** [comment]
Hello again @GabrielJones646!

can you past here the request (use curl or httpie) and the full response?

**Comment by GabrielJones646** [comment]
db request
```
[gjones@host ~]$ curl http://host:8080/foo
{ "_links" : { "self" : { "href" : "/foo"} , "curies" : [ ]} , "_id" : "foo" , "_db-props-cached" : false , "_size" : 3 , "_total_pages" : 1 , "_returned" : 3 , "_embedded" : { "rh:coll" : [ { "_links" : { "self" : { "href" : "/foo/migration_log"}} , "_id" : "migration_log"} , { "_links" : { "self" : { "href" : "/foo/raw_events"}} , "_id" : "raw_events"} , { "_links" : { "self" : { "href" : "/foo/raw_events_irr_test"}} , "_id" : "raw_events_irr_test"}]}}
```

collection request
```
[gjones@host ~]$ curl http://host:8080/foo/raw_events
{ "_links" : { "self" : { "href" : "/foo/raw_events"}} , "http status code" : 404 , "http status description" : "Not Found" , "message" : "Collection 'raw_events' does not exist"}
```

**Comment by ujibang** [comment]
I created a database with 3.0.7 configured to use the wired tiger storage engine.

I still couldn't reproduce the issue (and all test passed).

I suspect a bug in the mongodb java driver, at the end of the day the "collection does not exist" response happens if [DB.collectionExists()](http://api.mongodb.org/java/3.2/com/mongodb/DB.html#collectionExists-java.lang.String-) returns false.

So I refactored the code to check collection existence using the new java driver 3.2 [MongoDatabase.listCollectionNames()](http://api.mongodb.org/java/3.2/com/mongodb/client/MongoDatabase.html#listCollectionNames--)

Can you try with latest 1.1.3-SNAPSHOT version?

**Comment by GabrielJones646** [comment]
It looks like its trying to store and retrieve data from a _properties collection.  Is there not a way to run restheart in a purely read only mode on my mongoDB?

**Comment by GabrielJones646** [comment]
I was able to see the collection after I ran:

    db.getCollection("_properties").insert({_id: "_properties.raw_events"})

Is it a requirement that restheart have write access to my db in order to function?  This seems contrary to the [documentation](https://softinstigate.atlassian.net/wiki/display/RH/Installation+and+Setup#InstallationandSetup-7.3.MongoDBauthenticationwithjustenoughpermissionsauth-with-jep).

**Comment by ujibang** [comment]
Ah! With read only priviledge, the case where the db exists without the _properties collection is managed but not this case: 

Collection exists without _properties.<collname> => this happens when the collection has not been created by restheart.

I'll fix it by end of the week.

Well done, thanks

**Comment by GabrielJones646** [comment]
I didn't have the _properties collection.  It got added when I ran the insert.

**Comment by ujibang** [comment]
fixed in latest release 1.1.3
